apps/mobile/src/components/Feed/Posts/PostListCaption.tsx

```diff
@@ -1,17 +1,28 @@
-import { useMemo } from 'react';
+import { useMemo, useRef, useState } from 'react';
 import { View } from 'react-native';
 import { graphql, useFragment } from 'react-relay';
 
+import { GalleryTouchableOpacity } from '~/components/GalleryTouchableOpacity';
 import ProcessedText from '~/components/ProcessedText/ProcessedText';
 import { PostListCaptionFragment$key } from '~/generated/PostListCaptionFragment.graphql';
+import { GalleryElementTrackingProps } from '~/shared/contexts/AnalyticsContext';
 import { removeNullValues } from '~/shared/relay/removeNullValues';
 import { replaceUrlsWithMarkdownFormat } from '~/shared/utils/replaceUrlsWithMarkdownFormat';
 
 type Props = {
   feedPostRef: PostListCaptionFragment$key;
+  eventContext?: GalleryElementTrackingProps['eventContext'];
 };
 
-export function PostListCaption({ feedPostRef }: Props) {
+export function PostListCaption({ feedPostRef, eventContext }: Props) {
+  const [showAll, setShowAll] = useState(false);
+  const ref = useRef(null);
+  const toggleText = () => {
+    if (!showAll) {
+      setShowAll(!showAll);
+    }
+  };
```

Comment:
some small things:
1. let's wrap this in `useCallback`! we want to memoize things as much as possible on mobile for performance
2. `setShowAll` can actually accept a function that provides the previous state as the parameter, and you can `return` the next state. in general this should make it more functional and allow the dependecy list for the callback to be stateless:

```js
// good
const toggleText = useCallback(() => {
  if (!showAll) {
    setShowAll(!showAll);
  }
}, [showAll]);

// better
const toggleText = useCallback(() => {
  setShowAll((prev) => {
    if (!prev) return !prev;
    return prev;
  })
}, []); // no more dependency
```

now that i'm looking at this logic though: what is actually happening?
- if `showAll` is `true`, then nothing happens, and `showAll` remains `true`
- if `showAll` is `false`, then we `setShowAll(!false)`, which means we set it to `true`

so in every scenario we set the value to `true`, in which case we should really just make it:

```js
const toggleText = useCallback(() => {
  setShowAll(true)
}, []);
```

you might say: well, if `showAll` is already set to `true`, do we need to actively set it to `true` again? good thing is the react engine will notice that the value hasn't changed `true -> true` and will not trigger a re-render!

Reply:
done!